docs/melange-for-x-developers.md

```diff
@@ -0,0 +1,601 @@
+- `Marshal` is well supported in Js_of_ocaml, while Melange does not support it.
+- Libraries like `Unix` or `Str` are available in Js_of_ocaml but not in
+  Melange.
+- Js_of_ocaml supports sourcemaps, which Melange do not support yet (as of May
+  2023).
+
+On the upside, in Melange:
+
+- Consuming existing JavaScript packages might a bit easier in Melange, thanks
+  to its compilation model and the extensive availability of mechanisms to bind
+  to JavaScript code.
+- There is great support for some of the most used JavaScript libraries like
+  [ReactJS](https://github.com/reasonml/reason-react/) or GraphQL clients.
+- The generated JavaScript bundles are generally smaller.
+- The generated JavaScript code is generally more readable.
+- Straight-forward integration with JavaScript tooling like Webpack, NextJS,
+  etc. This is possible thanks to the 1 module <-> 1 JavaScript file compilation
+  model.
+
+## For ReScript developers
+
+As a project that was born originally from ReScript, Melange inherits a lot of
+its characteristics:
+
+- The compilation model involves compiling a single module into a single
+  JavaScript file.
+- The libraries provided by ReScript (Belt and Js) are available in Melange too.
+- The mechanisms provided for [communicating with JavaScript
+  code](communicate-with-javascript.md) are mostly the same.
+
+However, one of Melange’s goals is to maximize compatibility with the OCaml
+ecosystem. This goal translates into fundamental differences in how Melange and
+ReScript function from the perspective of both library authors and users.
+
+### Package manager
+
+ReScript projects rely exclusively on npm for all packages they depend on.
+Melange projects, on the other hand, will use opam for native packages, and npm
+for JavaScript ones. Melange package management is explained in detail in [the
+dedicated section](package-management.md).
+
+### Build system
+
+While ReScript has its own build system, originally based on Ninja, Melange
+defers to Dune as orchestrator of builds, as is explained in detail in [the
+corresponding section](build-system.md).
+
+The divergences caused by switching the build system have a lot of implications
+and nuances that are too complex to explain in this section, but some of the
+specific details have been discussed in [the OCaml
+forum](https://discuss.ocaml.org/t/ahrefs-is-now-built-with-melange/12107/3).
+
+### Source-based vs pre-built distribution
+
+While with ReScript every dependency can be downloaded with just npm, Melange
+projects will have to use opam and npm. This is a trade-off: on one hand, some
+Melange projects might need to include two package configuration files. But on
+the other hand they can benefit from opam’s source-based package distribution
+model for things like PPXs, linters, or any other OCaml tooling.
+
+By comparison, to consume any OCaml ecosystem tool in ReScript is more
+challenging. As npm is a package manager designed for an interpreted language,
```

Comment:
"there's no native toolchain available in rescript" should be mentioned somewhere. That's the actual limitation

Reply:
reworded in https://github.com/melange-re/documentation-site/pull/39/commits/dde8ae32d450e5e5dff9c2837f6970ec7bf916d2.